src/daft-functions-temporal/src/total.rs

```diff
@@ -0,0 +1,53 @@
+use common_error::{ensure, DaftError, DaftResult};
+use daft_core::{
+    datatypes::{DataType, Field},
+    prelude::{Schema, Series},
+    series::IntoSeries,
+};
+use daft_dsl::{
+    functions::{FunctionArgs, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+macro_rules! impl_total {
+    ($name:ident, $cast_method:ident) => {
+        paste::paste! {
+        #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+        pub struct $name;
+
+        #[typetag::serde]
+        impl ScalarUDF for $name {
+            fn name(&self) -> &'static str {
+                stringify!([ < $name:snake:lower > ])
+            }
+
+            fn evaluate(&self, inputs: daft_dsl::functions::FunctionArgs<Series>) -> DaftResult<Series> {
+                ensure!(inputs.len() == 1, format!("{} expects 1 argument", self.name()));
```

Comment:
you can specify one of the `DaftError` variants here. We usually do a `SchemaMismatchError` here.
```suggestion
ensure!(inputs.len() == 1, SchemaMismatch: "Expected 1 input, but received {}", inputs.len());
```